### PR TITLE
allow publishing airbyte-server to local maven repo

### PR DESCRIPTION
**Pull request by jrhizor**
This builds a jar that includes all dependencies.

If we need something the server doesn't include we can move this to a new project that depends on all of the relevant projects and call it `airbyte-cloud-export` or something.

If this is the direction we're going, I'll also add publication to a separate Maven repo as well.

I'm using a fat jar here to dodge the complications of publishing Maven dependencies for every subproject and making sure the dependency declarations for Maven match for each of them.

### CODE REVIEW

File: airbyte-server/build.gradle [hunk 1]
```diff
@@ -1,5 +1,26 @@
 plugins {
     id 'application'
+    id 'maven-publish'
+    id 'com.github.johnrengelman.shadow' version '6.1.0'
+}
+
+shadowJar {
+    zip64 true
+    mergeServiceFiles()
+    exclude 'META-INF/*.RSA'
```
Comment (davinchia) [inline airbyte-server/build.gradle:10]: what is this exclude doing?

File: airbyte-server/build.gradle [hunk 2]
```diff
@@ -1,5 +1,26 @@
 plugins {
     id 'application'
+    id 'maven-publish'
+    id 'com.github.johnrengelman.shadow' version '6.1.0'
```
Comment (davinchia) [inline airbyte-server/build.gradle:4]: shadow is back!!!

**Review by davinchia — APPROVED** [review]
I'm looking at the work that needs to be done and the following tasks will touch OSS:
1) migrating apps to use CloudSQL
2) webapp needs to hide certain UI elements in Cloud
3) making Workspace usable
4) Auth

I think 2 and 3 can be committed to the OSS repo, so 1 and 4 are the tasks that might require JARs.

4 is covered with the server fat jar. I think 1 might require us to publish everything that requires us to be able to inject CloudSql.

Commenting this to address your comments on needing a publish project. Not blocking since we can always add it later.

**Comment by ChristopheDuong** [comment]
In order to build the cloud-api, I would need more than just `airbyte-server` though...

For example, re-using:
- `airbyte-commons`
- `airbyte-db`
- `airbyte-config:models`
- `airbyte-config:persistence`

File: airbyte-server/build.gradle [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+}
+
+shadowJar {
+    zip64 true
+    mergeServiceFiles()
+    exclude 'META-INF/*.RSA'
+    exclude 'META-INF/*.SF'
+    exclude 'META-INF/*.DSA'
+    // Not stubbing this out adds 'all' to the end of the jar's name.
```
Comment (davinchia) [inline airbyte-server/build.gradle:13]: @jrhizor I had to add this so the names of the jars would line up. We either stub this out or we change the Cloud's dependency to be {module]-all.jar.

**Review by ChristopheDuong — APPROVED** [review]

File: airbyte-server/build.gradle (hunk 3 above)
Comment (jrhizor) [reply airbyte-server/build.gradle:13]: Hmm it was able to consume the `-all` jar for me properly for me. Not sure what made it required for you but it's fine either way.

File: airbyte-server/build.gradle (hunk 1 above)
Comment (jrhizor) [reply airbyte-server/build.gradle:10]: It removes signed certificates, which may conflict from some packages.

File: airbyte-server/build.gradle (hunk 3 above)
Comment (jrhizor) [reply airbyte-server/build.gradle:13]: Oh this is for the server Docker build?

**Review by cgardens — APPROVED** [review]
nice!

**Comment by jrhizor** [comment]
going to merge since it's failing due to a master failure only (gets past the main build step)

**Pull request merged**